crates/core/tedge_actors/src/test_helpers.rs

```diff
@@ -112,15 +110,82 @@ pub async fn assert_received_unordered<T, U>(
     expected: T,
 ) where
     T: IntoIterator,
-    U: From<T::Item>,
-    U: Debug + Eq + Hash,
+    T::Item: Debug + PartialEq<U>,
+    U: Debug + Eq,
+{
+    assert_received_matching(
+        messages,
+        |pat: &T::Item, msg: &U| pat == msg,
+        timeout,
+        expected,
+    )
+    .await
+}
+
+/// Check that at least one matching message is received for each pattern.
+///
+/// The messages can possibly be received in a different order or with interleaved messages.
+/// Returns early on `timeout` while waiting for the next message.
+///
+/// ```rust
+/// use crate::tedge_actors::{Builder, NoMessage, RuntimeError, ServiceConsumer, SimpleMessageBox, SimpleMessageBoxBuilder, test_helpers};
+///
+/// #[derive(Debug,Eq,PartialEq)]
+/// enum MyMessage {
+///    Foo(u32),
+///    Bar(u32),
+/// }
+///
+/// impl MyMessage {
+///     pub fn count(&self) -> u32 {
+///         match self {
+///             MyMessage::Foo(n) => *n,
+///             MyMessage::Bar(n) => *n,
+///         }
+///     }
+/// }
+///
+/// # #[tokio::main]
+/// # async fn main() -> Result<(),RuntimeError> {
+///
+/// use std::time::Duration;
+/// let mut receiver_builder = SimpleMessageBoxBuilder::new("Recv", 16);
+/// let sender_builder = SimpleMessageBoxBuilder::new("Send", 16).with_connection(&mut receiver_builder);
```

Comment:
Sorry it's not related to this PR, but I am confused `with_connection()` and `set_connection()`. Both API has the same implementation and exist. Now we need to use `with_connection()` right? (I saw `#[must_use]` flag)

Reply:
Ah non-reference vs reference.